exercises/risk-risiko.cpp

```diff
@@ -27,3 +27,97 @@
   M 3 vs 3 => blue win
   O 2 vs 1 => red win
 */
+
+#include <iostream>
+#include <ctime>
+#include <cstdlib>
+
+using namespace std;
+
+class giocatore {
+  public:
+  giocatore() : N(0),M(0),O(0) {}
+  void roll();
+  protected:
+  virtual void print() {};
+  int O;
+  int M;
+  int N;
+};
+
+class attacco;
+class difesa;
+void battaglia(attacco a,difesa d);
+
+class attacco : public giocatore {
+  public:
+  friend void battaglia(attacco a, difesa d);
+  attacco() : giocatore() {}
+  private:
+  void print();
+};
+
+class difesa : public giocatore {
+  public:
+  friend void battaglia(attacco a, difesa d);
+  difesa() : giocatore() {}
+  private:
+  void print();
+};
+
+int main() {
+  srand(time(NULL));
+  attacco red;
+  difesa blue;
+  red.roll();
+  blue.roll();
+  battaglia(red,blue);
+  return 0;
+}
+
+void battaglia(attacco a,difesa d) {
+  cout<<"  R    B"<<endl;
+  cout<<"N "<<a.N<<" vs "<<d.N<<" => ";
+  if (d.N>=a.N) cout<<"blue win";
+  else cout<<"red win";
+  cout<<endl;
```

Comment:
potresti utilizzare l'operatore ternario e scrivere il codice in maniera più compatta
```c++
cout  << d.N >= a.N ? "blue" : "red" << " win" << endl;
```
Oppure...

```c++
string winner = d.N >= a.N ? "blue" : "red";
cout  << winner << " win" << endl;
```